graph/patterns/longRunningOperations.md

```diff
@@ -0,0 +1,192 @@
+web style as other interactive APIs. It also needs to support checking on the status and/or
+being notified asynchronously of the results.
+
+## Solution
+
+The solution is to model the API as a synchronous service that returns a
+resource that represents the eventual completion or failure of a long running
+operation.
+
+There are two flavors of this solution:
+
+- The returned resource is the targeted resource and includes the status of
+    the operation. This pattern is often called RELO (resource-based
+    long running operation).
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="RELO.gif" alt="The status monitor LRO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+- The returned resource is a new API resource called 'Stepwise Operation' and
+    is created to track the status. This LRO solution is similar to the concept
+    of Promises or Futures in other programming languages.
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="LRO.gif" alt="The status monitor LRO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+The RELO pattern is the preferred pattern for long running operations and should be
+used wherever possible. The pattern avoids complexity, and consistent resource
+presentation makes things simpler for our users and tooling chain.
+
+In general, Microsoft Graph API guidelines for LRO follow [Microsoft REST API
+Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations).
+
+A single deviation from the base guidelines is that Microsoft Graph API standards require that you use the following response headers:
+
+- The Content-Location header indicates the location of a RELO resource.
+  - The API response says the targeted resource is being created by returning a 201 status code and the resource URI is provided in the Content-Location header, but the response indicates that the request is not completed by including "Provisioning" status.
+
+- The Location header indicates the location of a new stepwise operation LRO resource.
+  - The API response says the operation resource is being created at the URL provided in the Location header and indicates that the request is not completed by including a 202 status code.
+
+## When to use this pattern
+
+Any API call that is expected to take longer than 1 second in the 99th percentile should use the long running operations pattern.
+
+How do you select which flavor of LRO pattern to use? An API designer can follow these
+heuristics:
+
+1.  If a service can create a resource with a minimal latency and continue
+    updating its status according to the well-defined and stable state
+    transition model until completion, then the RELO model is the best choice.
+
+2.  Otherwise, a service should follow the Stepwise Operation pattern. 
+ 
+
+## Issues and considerations
+
+- One or more clients MUST be able to monitor and operate on the same resource at the same time.
```

Comment:
Should we use API consumer instead of client here, since the consumers of an API may themselves be services

```suggestion
- One or more API consumers MUST be able to monitor and operate on the same resource at the same time.
```